go/ubirch/ubirch_sim.go

```diff
@@ -180,6 +180,17 @@ func (p *Protocol) response(code uint16) (string, uint16, error) {
 	return data, code, nil
 }
 
+func (p *Protocol) GetIMSI() (string, error) {
+	imsi, err := p.Send("AT+CIMI")
+	if err != nil {
+		return "", err
+	}
+	if imsi[0] == "ERROR" {
+		return "", errors.New("no IMSI available")
+	}
+	return imsi[0], nil
```

Comment:
Possibly also check imsi is sane (maximum length and that it only contains numbers)? We might get almost anything from the SIM if something went wrong in the previous command/communication.

Reply:
yes! added todo